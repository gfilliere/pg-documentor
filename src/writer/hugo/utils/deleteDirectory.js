import fs from 'fs';
import rimraf from 'rimraf';

export default directory =>
  new Promise((resolve, reject) => {
    rimraf(directory, fs, err => {
      if (err) {
        return reject(err);
      }
      return resolve();
    });
  });
