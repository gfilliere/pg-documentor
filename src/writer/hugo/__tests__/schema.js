import { movieSchema } from '../../../fixtures/sampleSchema';
import generateSchema from '../schema';
import deleteDirectory from '../utils/deleteDirectory';

jest.mock('fs');

describe('generateSchema', () => {
  const fs = require('fs'); // eslint-disable-line global-require
  beforeAll(() => {
    fs.mkdirSync('/memory');
    fs.mkdirSync('/memory/hugo');
  });

  beforeEach(async () => {
    await deleteDirectory('/memory/hugo/movie');
  });

  test('a file for each table is created', async () => {
    expect.assertions(2);
    await generateSchema(movieSchema, '/memory/hugo');

    expect(fs.existsSync('/memory/hugo/movie/movie.md', 'utf8')).toBe(true);
    expect(fs.existsSync('/memory/hugo/movie/movie_has_image.md', 'utf8')).toBe(
      true,
    );
  });

  test('an _index.md is generated and valid', async () => {
    expect.assertions(1);
    await generateSchema(movieSchema, '/memory/hugo');

    expect(
      fs.readFileSync('/memory/hugo/movie/_index.md', 'utf8'),
    ).toMatchSnapshot();
  });
});
