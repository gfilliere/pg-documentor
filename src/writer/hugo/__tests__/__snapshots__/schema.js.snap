// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateSchema an _index.md is generated and valid 1`] = `
"+++
title = \\"movie\\"
description = \\"Content of the movie schema\\"
alwaysopen = false
+++


- movie
- movie_has_image
    "
`;
